### PR TITLE
[CI] Add cleanup for workflows/sycl_gen_test_matrix.yml

### PR DESCRIPTION
**Pull request by aelovikov-intel**
It's generally good when the task doesn't leave any leftovers after itself. That becomes even more important when using self-hosted runners where leftovers aren't cleared by the container destruction (if one is used at all).

This is also kind of related to the root/actions/checkout/lint issue that (I believe with about 90% confidence) is not caused by the actions/checkout. Instead, it's the difference under which user the job is executed due to the containers usage.

In a nutshell, I'm trying to move our ubuntu-* tasks onto local runners and this PR is one of several change I'm making/made. I'm currently using the cuda runner because we can get by with tasks not cleaning after themselves as no real jobs are being scheduled there. However, I'm working on implementing all the necessary cleanups so that we can schedule them on any Linux runner and returning the cuda runner back to the pool where we can run CUDA testing on it.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
Tested manually in https://github.com/intel/llvm/actions/runs/5213273226/jobs/9408036060 via observing filesystem on the self-hosted runner where the job is executed. Note that manual cleanup will be needed after this is merged in because all other current tasks run the job without cleanup and leave files after themselves.

**Comment by aelovikov-intel** [comment]
> Any description with the explanation about why this change is needed?

It's generally good when the task doesn't leave any leftovers after itself. 

This is also kind of related to the `root`/`actions/checkout`/`lint` issue that (I believe with about 90% confidence) is *not* caused by the `actions/checkout`. Instead, it's the difference under which user the job is executed due to the containers usage.

In a nutshell, I'm trying to move our `ubuntu-*` tasks onto local runners. I'm currently using the `cuda` runner because we can get by with tasks not cleaning after themselves as no real jobs are being scheduled there. However, I'm working on implementing all the necessary cleanups so that we can schedule them on any `Linux` runner and returning the `cuda` runner back to the pool where we can run CUDA testing on it.

**Comment by bader** [comment]
> > Any description with the explanation about why this change is needed?
> 
> It's generally good when the task doesn't leave any leftovers after itself.
> 
> This is also kind of related to the `root`/`actions/checkout`/`lint` issue that (I believe with about 90% confidence) is _not_ caused by the `actions/checkout`. Instead, it's the difference under which user the job is executed due to the containers usage.
> 
> In a nutshell, I'm trying to move our `ubuntu-*` tasks onto local runners. I'm currently using the `cuda` runner because we can get by with tasks not cleaning after themselves as no real jobs are being scheduled there. However, I'm working on implementing all the necessary cleanups so that we can schedule them on any `Linux` runner and returning the `cuda` runner back to the pool where we can run CUDA testing on it.

Please, put this text to PR description.

**Comment by aelovikov-intel** [comment]
> Please, put this text to PR description.

Done, with some minor additions.

**Comment by bader** [comment]
> @bader , do you expect anything else from me here?

This looks fine, but we should be consistent in applying our rules. Make sure the same logic is applied to #9797 when it's ready for review.

**Review by bader — APPROVED** [review]
